onnxruntime/core/framework/prepacked_weights_container.h

```diff
@@ -66,4 +73,163 @@ class PrepackedWeightsContainer final {
   std::unordered_map<std::string, PrePackedWeights> prepacked_weights_map_;
 };
 
+/// <summary>
+/// This class has a dual purpose.
```

Comment:
What's the reason we don't create a PrepackedWeightsContainer when loading from disk? Having a class called `PrepackedForSerialization` in places like session_state_utils when we're reading from it is confusing. 

Or can we refactor to have a base class with shared functionality and derived classes for the 'populate to save' and 'provide access to weights memory mapped from disk' usages?

Reply:
Simplified container.